components/resources/quizzes/careerQuiz/career-quiz.tsx

```diff
@@ -87,21 +124,58 @@ const CareerQuiz = () => {
   };
 
   const handleOptionClick = (option: QuizOptionViewState) => {
-    const selectedQuizOption = quizViewState.questions.map((question) => ({
-      ...question,
-      options: question.options.map((questionOption) =>
-        questionOption.name === option.name
-          ? { ...questionOption, isSelected: true }
-          : questionOption
-      ),
-    }));
-
-    const updatedQuizViewState = {
-      ...quizViewState,
-      questions: selectedQuizOption,
-    };
+    const currentQuestion =
+      quizViewState.questions[quizViewState.currentQuestion];
+    const selectedOptions = currentQuestion.options.filter(
+      (option) => option.isSelected
+    );
+    const maxSelection =
+      quizData.questions[quizViewState.currentQuestion].maxSelections;
+
+    if (maxSelection === 1) {
```

Comment:
i think this logic is too long and a little over-complicated.. i played around with some code and got a simpler approach very close to working

check my suggestion and lmk if i'm overlooking something: 
- move the optional maxSelections prop to the QuizQuestionViewState type
- change how you're initializing the quiz view state so it accounts for this type change
- now you can access maxSelections in the skillSelections prop
- change the answer selection from a div to a button so you have access to the disabled prop
- store the number of questions selected in a local state and increment / decrement inside your handle click
```
const handleClick = (option: QuizOptionViewState) => {
    handleOptionClick(option);
    option.isSelected
      ? setNumOptionsSelected((prev) => prev - 1)
      : setNumOptionsSelected((prev) => prev + 1);
  };
```
- disable the button if num questions selected greater than max selections  
```
disabled={numOptionsSelected >= maxSelections && !option.isSelected}
```

only thing i didn't fully figure out was resetting the state when you switch to a different question